cloud-inquisitor/aws_s3_resource.go

```diff
@@ -43,7 +45,7 @@ type AWSS3Storage struct {
 	Tags map[string]string
 }
 
-func (s *AWSS3Storage) NewFromEventBus(event events.CloudWatchEvent) error {
+func (s *AWSS3Storage) NewFromEventBus(event events.CloudWatchEvent, logger *log.Logger) error {
```

Comment:
Do you think it might make the code neater if we make `logger` as singleton?

Reply:
@riot-jetaylor +1 to this.  Passing the logger around seems cumbersome. 

Is there a way to lift it up to either a property of the resource or something above that?  I don't know how unique each logger needs to be per run.

Reply:
@HTTP500 @rnikoopour I have moved the logger to be initialized with the resource instead of in the step. This allows the logger to be owned by the resource and used in method calls without being passed in. However, this means well need more verbose logging within each resource to make up for logging within the step.